### PR TITLE
Update docs due to #741

### PR DESCRIPTION
**Pull request by sunlark**
1. ReactionCore.Locale.currency and ReactionCore.Locale.shopCurrency were explained.

2. added description of new Shop currencies option 'where'

### CODE REVIEW

**Comment by sunlark** [comment]
Accidentally added commits here instead of new branch. 

This PR depends on reactioncommerce/reaction#741

1. removed incorrect line of text inside Velocity/Integration (duplicate from Velocity/End to End)
2. Factories description extended and updated
3. Faker description extended and updated
4. added description of some non-obvious fields in Product and ProductVariant schemas

**Comment by zenweasel** [comment]
@aaronjudd My mistake for not putting notes on this, do you remember why we blocked this? Should I fix up the couple of grammar errors and unblock?

**Comment by aaronjudd** [comment]
I think it was blocked because #741 hadn't been merged yet? Looks like unblocking and cleanup should take care of this.

File: developer/testing/velocity.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ As such, our testing falls into two locations:
    You enable these tests by uncommenting the velocity testing packages in `.meteor/packages`.
 
 ## Integration
-   Tests for the individual reaction packages, are respectively located in `/tests/jasmine/server/integration`. Each package should cover tests  these tests cover UIX testing. These are relatively fragile tests as they are only testing core theme and UIX.
+   Tests for the individual reaction packages, are respectively located in `/tests/jasmine/server/integration`. Each package should cover tests  these tests cover UIX testing.
```
Comment (zenweasel) [inline developer/testing/velocity.md]: I think that last sentence (not yours but it's a little confusing) should read something like:
`Each package should have tests that cover their own functionality. These tests only cover UIX testing.`

File: developer/architecture/schemas.md [hunk 2]
```diff
@@ -110,3 +110,17 @@ ReactionCore.Schemas.PaypalPackageConfig = new SimpleSchema([
   }
 ]);
 ```
+### Description of some non-obvious fields
+#### Product
+- **ancestors: Array** ancestors array for product is always clean for now
```
Comment (zenweasel) [inline developer/architecture/schemas.md]: I'm not clear on "already clean" would mean? Always empty?

File: developer/testing/velocity.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ As such, our testing falls into two locations:
    You enable these tests by uncommenting the velocity testing packages in `.meteor/packages`.
```
Comment (zenweasel) [inline developer/testing/velocity.md]: This line is no longer true as these packages are not in the packages files any longer.

**Comment by zenweasel** [comment]
@sunlark  Thanks for the PR. If we could make these few changes and resolve the conflicts we can merge this in.

File: developer/architecture/schemas.md (hunk 2 above)
Comment (newsiberian) [inline developer/architecture/schemas.md]: Ah, yes. For product it is aways empty with current logic.

**Comment by sunlark** [comment]
Hello. I've made the changes you asked.

**Pull request merged**